website/docs/configuration/overview.html.md

```diff
@@ -0,0 +1,182 @@
+---
+layout: "language"
+page_title: "Configuration Language"
+sidebar_current: "docs-config-index"
+description: |-
+  Terraform uses text files to describe infrastructure and to set variables.
+  These text files are called Terraform _configurations_ and are
+  written in the Terraform language.
+---
+
+# Configuration Language
+
+-> **Note:** This page is about Terraform 0.12 and later. For Terraform 0.11 and
+earlier, see
+[0.11 Configuration Language](../configuration-0-11/index.html).
+
+> **Hands-on:** Try the [Terraform: Get Started](https://learn.hashicorp.com/collections/terraform/aws-get-started?utm_source=WEBSITE&utm_medium=WEB_IO&utm_offer=ARTICLE_PAGE&utm_content=DOCS) collection on HashiCorp Learn.
+
+Terraform uses its own configuration language, designed to allow concise
+descriptions of infrastructure. The Terraform language is declarative,
+describing an intended goal rather than the steps to reach that goal.
+
+## Resources and Modules
+
+The main purpose of the Terraform language is declaring [resources](./resources.html).
+All other language features exist only to make the definition of resources
+more flexible and convenient.
+
+A group of resources can be gathered into a [module](./modules.html),
+which creates a larger unit of configuration. A resource describes a single
+infrastructure object, while a module might describe a set of objects and the
+necessary relationships between them in order to create a higher-level system.
+
+A _Terraform configuration_ consists of a _root module_, where evaluation
+begins, along with a tree of child modules created when one module calls
+another.
```

Comment:
So, preamble to this and all the following comments on this page: This language overview isn't new text that rode in on this PR, it's all stuff that we already had at configuration/index.html. So I'm down to reconsider any of it, but it's not something I just added today, and it's not something where I made all the decisions! Just in case you're wondering why it seems different from all the other new pages. :) 

Anyway IMO this is constructing the hierarchy of "resources -> module (of resources) -> configuration (of modules)".

Reply:
Ah, this makes more sense. I don't want to hold this PR up for this page, so maybe the thing to do is table changes here until we have a more break-y PR, and maybe think though this page content some more? Anything that's easy it would be nice to do, but since this isn't a new page I can see how it might be out of scope. 

Is this a new page that will need aa redirect if we scrap it later? That's the only reason to pin down the intention for this page now.